### PR TITLE
Fixes #6275: Redirecting /user/<userId> to /user/<username>

### LINKED ISSUE #6275: Redirect user profile URLs using user id to the canonical variant using username
### Describe the problem and steps to reproduce it:
- Visit https://addons.mozilla.org/en-US/firefox/user/4757633/

### What happened?
- The page loads directly

### What did you expect to happen?
- The client should be redirect to the canonical URL using the username instead of the user id, which is https://addons.mozilla.org/en-US/firefox/user/mozilla/

### Anything else we should know?
- We do that for add-ons already: https://addons.mozilla.org/en-US/firefox/addon/954390/ redirects to https://addons.mozilla.org/en-US/firefox/addon/facebook-container/

### PR DESCRIPTION
**Pull request by suhailsinghbains**
Fixes #6275
On visiting https://addons.mozilla.org/en-US/firefox/user/4757633/ , the page is redirected to https://addons.mozilla.org/en-US/firefox/user/mozilla/

No pictures could be attached, as a Video demonstrating would be appropriate.
Thanks @willdurand for being my mentor.

### CODE REVIEW

File: src/amo/pages/UserProfile/index.js [hunk 1]
```diff
@@ -112,6 +113,16 @@ export class UserProfileBase extends React.Component<InternalProps> {
         }),
       );
     }
+    let idOrNameCheck = { ...location}.pathname.split('/')[4];
```
Comment (willdurand) [inline src/amo/pages/UserProfile/index.js]: hu oh, what's that? Why not using `location` directly? In any case I don't think you need the `location`, there should be a `match` prop that does that (i.e. extracting URL parameters).

File: src/amo/pages/UserProfile/index.js [hunk 2]
```diff
@@ -112,6 +113,16 @@ export class UserProfileBase extends React.Component<InternalProps> {
...
+    if(/^[0-9]+$/.test(idOrNameCheck)){
```
Comment (willdurand) [inline src/amo/pages/UserProfile/index.js]: I think you can use `\d+`

File: src/amo/pages/UserProfile/index.js [hunk 3]
```diff
@@ -112,6 +113,16 @@ export class UserProfileBase extends React.Component<InternalProps> {
...
+      const newUrl= { ...user };
```
Comment (willdurand) [inline src/amo/pages/UserProfile/index.js]: You can remove that and use `user.username` in the `url:` value.

Comment (suhailsinghbains) [reply src/amo/pages/UserProfile/index.js]: I'm getting this error and hence, does not redirect. I just cannot use any property of 'user'.
UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'username' of undefined

Comment (willdurand) [reply src/amo/pages/UserProfile/index.js]: ah, I see. Yes, because we need to fetch the user first. So we need the user to send the server redirect.

**Comment by willdurand** [comment]
@suhailsinghbains hi, do you need any help?

**Comment by suhailsinghbains** [comment]
> 
> 
> @suhailsinghbains hi, do you need any help?

Yes please, I tried yarn test, every test passed, but, here travis is giving error.

**Comment by suhailsinghbains** [comment]
The yarn amo:dev project works perfectly(redirection)
But, I couldn't figure out how to satisfy the test. I had been trying for past 3 hours. Please help me @willdurand

**Comment by willdurand** [comment]
>  But, I couldn't figure out how to satisfy the test. I had been trying for past 3 hours. Please help me @willdurand

That's because we have to update some other tests, because the current logic uses the `username` as default parameter. This requires some changes and I filed #6526 for that (I'm going to work on it first, then you'll be able to rebase this PR on top of my patch for #6526)

**Comment by suhailsinghbains** [comment]
Ohh, Thank God, I was thinking that I was disturbing you for such small things :)

**Comment by muffinresearch** [comment]
@suhailsinghbains hi - have you had a chance to take another look at this?

**Comment by suhailsinghbains** [comment]
@muffinresearch Hi, I got busy with the exams, so, didn't get time to go through again.
>That's because we have to update some other tests, because the current logic uses the username as default parameter. This requires some changes and I filed #6526 for that (I'm going to work on it first, then you'll be able to rebase this PR on top of my patch for #6526)

This issue(#6526 ) is open though, 
Still, I'll have a look :)

**Comment by willdurand** [comment]
@suhailsinghbains thanks a bunch for your work here. Unfortunately, things have changed a bit and also a lot of other things happened in the meantime, so I am not able to accept this PR anymore (or to help you iterate on it).

You can find other issues by looking at the [opened `contrib: welcome` issues](https://github.com/mozilla/addons-frontend/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22contrib%3A+welcome%22). Those issues should be safer (we won't reject your PRs) 😛

**Pull request closed without merging**